AEPCore/Sources/core/MobileCoreInitializer.swift

```diff
@@ -0,0 +1,108 @@
+    public internal(set) var classFinder = ClassFinder.classes(conformToProtocol:)
+#else
+    // Function to return list of classes; (Protocol) -> [AnyClass]
+    private let classFinder = ClassFinder.classes(conformToProtocol:)
+#endif
+
+#if DEBUG
+    public internal(set) static var shared = MobileCoreInitializer()
+#else
+    static let shared = MobileCoreInitializer()
+#endif
+
+    @available(iOSApplicationExtension, unavailable)
+    @available(tvOSApplicationExtension, unavailable)
+    func initialize(options: InitOptions, _ completion: (() -> Void)? = nil) {
+
+        if initialized.incrementAndGet() != 1 {
+            Log.debug(label: LOG_TAG, "initialize - ignoring as it was already called.")
+            return
+        }
+
+        if options.appGroup != nil {
+            MobileCore.setAppGroup(options.appGroup)
+        }
+
+        // Register Extensions, call configureWithAppId from callback
+        DispatchQueue.global().async {
+            let classList = self.classFinder(Extension.self)
+            let filteredClassList = classList.filter { $0 !== AEPCore.EventHubPlaceholderExtension.self && $0 !== AEPCore.Configuration.self }.compactMap { $0 as? NSObject.Type }
+            MobileCore.registerExtensions(filteredClassList) {
+
+                if let appId = options.appId {
+                    MobileCore.configureWith(appId: appId)
+                } else if let filePath = options.filePath {
+                    MobileCore.configureWith(filePath: filePath)
+                }
+
+                // If lifecycleAutomaticTracking flag is false, set lifecycle notification listeners
+                if options.lifecycleAutomaticTrackingEnabled == true {
+                    var usingSceneDelegate = false
+                    if #available(iOS 13.0, tvOS 13.0, *) {
+                        let sceneDelegateClasses = self.classFinder(UIWindowSceneDelegate.self)
+                        if !sceneDelegateClasses.isEmpty {
+                            usingSceneDelegate = true
+                        }
+                    }
+                    self.setupLifecycle(usingSceneDelegate: usingSceneDelegate, additionalContextData: options.lifecycleAdditionalContextData)
+                    Log.trace(label: self.LOG_TAG, "initialize - automatic lifecycle tracking enabled for \(usingSceneDelegate ? "UIScene" : "UIApplication").")
+                } else {
+                    Log.trace(label: self.LOG_TAG, "initialize - automatic lifecycle tracking disabled.")
+                }
+
+                completion?()
+            }
+        }
+    }
+
+    @available(iOSApplicationExtension, unavailable)
+    @available(tvOSApplicationExtension, unavailable)
+    private func setupLifecycle(usingSceneDelegate: Bool, additionalContextData: [String: Any]? = nil) {
+        if usingSceneDelegate {
+            MobileCore.lifecycleStart(additionalContextData: additionalContextData)
```

Comment:
Is this correct? Shouldn't we trigger lifecycleStart only when the app is foreground?